integration/fixture_test.go

```diff
@@ -0,0 +1,177 @@
+//+build integration
+
+package integration
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"go/build"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+var packageDir string
+
+const namespaceFlag = "-n=tilt-integration"
+
+func init() {
+	pkg, err := build.Default.Import("github.com/windmilleng/tilt/integration", ".", build.FindOnly)
+	if err != nil {
+		panic(fmt.Sprintf("Could not find integration test source code: %v", err))
+	}
+
+	packageDir = pkg.Dir
+}
+
+type fixture struct {
+	t      *testing.T
+	ctx    context.Context
+	cancel func()
+	logs   *bytes.Buffer
+	cmds   []*exec.Cmd
+}
+
+func newFixture(t *testing.T, dir string) *fixture {
+	err := os.Chdir(filepath.Join(packageDir, dir))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	f := &fixture{
+		t:      t,
+		ctx:    ctx,
+		cancel: cancel,
+		logs:   bytes.NewBuffer(nil),
+	}
+	f.CreateNamespaceIfNecessary()
+	f.ClearNamespace()
+	return f
+}
+
+func (f *fixture) Curl(url string) (string, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return "", errors.Wrap(err, "Curl")
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		f.t.Errorf("Error fetching %s: %s", url, resp.Status)
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", errors.Wrap(err, "Curl")
+	}
+	return string(body), nil
+}
+
+func (f *fixture) CurlUntil(ctx context.Context, url string, expectedContents string) {
+	actualContents := ""
+	for true {
```

Comment:
Could this be shortened to just `for {`?

Reply:
done